services/common/auth_tools.py

```diff
@@ -0,0 +1,369 @@
+    ADMIN = "admin"
+
+
+class RESOURCE_TYPE:
+    USER = "user"
+    RSU = "rsu"
+    INTERSECTION = "intersection"
+    ORGANIZATION = "organization"
+
+
+class UserInfo:
+    def __init__(self, token_user_info: dict):
+        logging.warning("Token User Info: " + str(token_user_info))
+        self.email = token_user_info.get("email")
+        self.organizations: dict[str, ORG_ROLE_LITERAL] = {
+            org["org"]: org["role"]
+            for org in token_user_info.get("cvmanager_data", {}).get(
+                "organizations", []
+            )
+        }
+        self.super_user = (
+            token_user_info.get("cvmanager_data", {}).get("super_user") == "1"
+        )
+        self.first_name = token_user_info.get("given_name")
+        self.last_name = token_user_info.get("family_name")
+        self.name = token_user_info.get("name")
+
+    # This method is exposed to users. It should not include any confidential information.
+    def to_dict(self):
+        return {
+            "email": self.email,
+            "organizations": [
+                {"org": name, "role": role} for name, role in self.organizations.items()
+            ],
+            "super_user": self.super_user,
+            "first_name": self.first_name,
+            "last_name": self.last_name,
+            "name": self.name,
+        }
+
+    def __repr__(self):
+        return json.dumps(self, default=lambda o: o.__dict__, sort_keys=True, indent=4)
+
+
+ENVIRON_USER_KEY = "user"
+
+
+class EnvironNoAuth:
+    def __init__(self):
+        self.user_info = None
+        self.organization = None
+        self.role = None
+
+
+class EnvironWithoutOrg:
+    def __init__(self, user_info: UserInfo):
+        self.user_info = user_info
+        self.organization = None
+        self.role = None
+
+
+class EnvironWithOrg(EnvironWithoutOrg):
+    def __init__(self, user_info: UserInfo, organization: str, role: ORG_ROLE_LITERAL):
+        self.user_info = user_info
+        self.organization = organization
+        self.role = role
+
+
+####################################### Restrictions By Organization #######################################
+def get_rsu_dict_for_org(organizations: list[str]) -> dict[str, str]:
+    if not organizations:
+        return {}
+    allowed_orgs_str = ", ".join(f"'{org}'" for org in organizations)
+    query = (
+        "SELECT rsu.ipv4_address::text AS ipv4_address "
+        "FROM public.rsus rsu "
+        "JOIN public.rsu_organization AS rsu_org ON rsu_org.rsu_id = rsu.rsu_id "
+        "JOIN public.organizations AS org ON org.organization_id = rsu_org.organization_id "
+        f"WHERE org.name = ANY (ARRAY[{allowed_orgs_str}])"
+    )
+
+    logging.debug(f'Executing query: "{query};"')
+    data = pgquery.query_db(query)
+
+    return {rsu["ipv4_address"]: rsu["ipv4_address"] for rsu in data}
+
+
+def check_rsu_with_org(rsu_ip: str, organizations: list[str]) -> bool:
+    if not organizations:
+        return {}
```

Comment:
chore(blocking): the return type of the method is bool, so we should return a bool not an empty dict
```suggestion
        return False
```